src/binary/writer.rs

```diff
@@ -458,72 +632,31 @@ impl<W: Write> BinarySystemWriter<W> {
     }
 
     /// Writes an Ion decimal with the specified value.
-    pub fn write_decimal(&mut self, value: &Decimal) -> IonResult<()> {
+    fn write_decimal(&mut self, value: &Decimal) -> IonResult<()> {
         self.write_scalar(|enc_buffer| {
             let _ = enc_buffer.encode_decimal_value(value);
             Ok(())
         })
     }
 
-    /// Writes an Ion decimal with the specified value.
-    pub fn write_big_decimal(&mut self, value: &BigDecimal) -> IonResult<()> {
-        self.write_scalar(|enc_buffer| {
-            // TODO: Currently clones. Revisit this API, similar to the comment
-            // on `write_datetime`.
-            let decimal: Decimal = value.clone().into();
-            let _ = enc_buffer.encode_decimal_value(&decimal);
-            Ok(())
-        })
-    }
-
     /// Writes an Ion timestamp with the specified value.
-    #[deprecated(
-        since = "0.6.1",
-        note = "Please use the `write_timestamp` method instead."
-    )]
-    pub fn write_datetime(&mut self, value: &DateTime<FixedOffset>) -> IonResult<()> {
-        self.write_scalar(|enc_buffer| {
-            // TODO: Currently this clones the Chrono type so we can make a
-            // Timestamp. In #273 we discuss traits that will avoid this clone.
-            // However, this API (`write_datetime`) is probably also not quite
-            // right.
-            let timestamp: Timestamp = (*value).into();
-            let _ = enc_buffer.encode_timestamp_value(&timestamp)?;
-            Ok(())
-        })
-    }
-
-    /// Writes an Ion timestamp with the specified value.
-    pub fn write_timestamp(&mut self, value: &Timestamp) -> IonResult<()> {
+    fn write_timestamp(&mut self, value: &Timestamp) -> IonResult<()> {
         self.write_scalar(|enc_buffer| {
             let _ = enc_buffer.encode_timestamp_value(value)?;
             Ok(())
         })
     }
 
-    pub fn write_symbol_id(&mut self, symbol_id: SymbolId) -> IonResult<()> {
-        self.write_scalar(|enc_buffer| {
-            const SYMBOL_BUFFER_SIZE: usize = mem::size_of::<u64>();
-            let mut buffer = [0u8; SYMBOL_BUFFER_SIZE];
-            let mut writer = io::Cursor::new(&mut buffer).writer();
-            let encoded_length = DecodedUInt::write_u64(&mut writer, symbol_id as u64)?;
-
-            let type_descriptor: u8;
-            if encoded_length <= MAX_INLINE_LENGTH {
-                type_descriptor = 0x70 | encoded_length as u8;
-                enc_buffer.push(type_descriptor);
-            } else {
-                type_descriptor = 0x7E;
-                enc_buffer.push(type_descriptor);
-                VarUInt::write_u64(enc_buffer, encoded_length as u64)?;
+    fn write_symbol<A: AsRawSymbolTokenRef>(&mut self, value: A) -> IonResult<()> {
+        match value.as_raw_symbol_token_ref() {
```

Comment:
`write_symbol_id` is still available on the `RawBinaryWriter`, but the trait method `write_symbol` accepts `A: AsSymbolTokenRef` instead of `SymbolId`.